### PR TITLE
Added delete url functionality

### LINKED ISSUE #14: Add Delete url functionality

### LINKED ISSUE #16: make custom-alias and expiry data optional

### PR DESCRIPTION
**Pull request by Dhruv88**
Resolves #14  
Resolves #16

### CODE REVIEW

**Review by shashwatj07 — CHANGES_REQUESTED** [review]
update readme

File: constants.go [hunk 1]
```diff
@@ -1,6 +1,10 @@
...
 // The host URL
-const HOST_URL = "http://localhost:8080/" 
+const HOST_URL = "http://localhost:8080/"
+
 // The host port
-const HOST_PORT = "localhost:8080" 
+const HOST_PORT = "localhost:8080"
+
+// Default Expiry days
+const DEFAULT_EXPIRY_DATE = 30
```
Comment (shashwatj07) [inline constants.go:10]: please make it ```DEFAULT_VALIDITY_DAYS```

File: handlers.go [hunk 2]
```diff
@@ -100,6 +121,19 @@ func Redirect(c *gin.Context) {
...
+func DeleteUrl(c *gin.Context) {
+	shortUrl := c.Param("shortUrl")
+	error := repo.Delete(shortUrl)
+	if error != nil {
+		log.Println(error)
+		c.AbortWithStatus(500)
+	} else {
+		c.Status(204)
+	}
+
```
Comment (shashwatj07) [inline handlers.go:134]: remove blank line

**Review by shashwatj07 — CHANGES_REQUESTED** [review]

**Comment by shashwatj07** [comment]
"expiry date" should be modified to "validity" everywhere

File: constants.go (hunk 1 above)
Comment (Dhruv88) [reply constants.go]: Done

File: handlers.go (hunk 2 above)
Comment (Dhruv88) [reply handlers.go]: Done

**Comment by Dhruv88** [comment]
> "expiry date" should be modified to "validity" everywhere

I have done this in most places except a few which are
line 6 in entity.go and lines 45-47 in repository.go where it is actually the UNIX timestamp of the expiry date and not just a validity period.

**Review by shashwatj07 — APPROVED** [review]

**Pull request merged**